apps/meteor/server/models/raw/Reads.ts

```diff
@@ -0,0 +1,62 @@
+import type { Reads, IUser, IMessage, RocketChatRecordDeleted } from '@rocket.chat/core-typings';
+import type { IReadsModel } from '@rocket.chat/model-typings';
+import type { Collection, Db, IndexDescription, UpdateResult } from 'mongodb';
+
+import { BaseRaw } from './BaseRaw';
+
+export class ReadsRaw extends BaseRaw<Reads> implements IReadsModel {
+	constructor(db: Db, trash?: Collection<RocketChatRecordDeleted<Reads>>) {
+		super(db, 'reads', trash);
+	}
+
+	protected modelIndexes(): IndexDescription[] {
+		return [{ key: { tmid: 1, userId: 1 }, unique: true }];
+	}
+
+	async findOneByUserIdAndThreadId(userId: IUser['_id'], tmid: IMessage['_id']): Promise<Reads | null> {
+		return this.findOne({ userId, tmid });
+	}
+
+	getMinimumLastSeenByThreadId(tmid: IMessage['_id']): Promise<Reads | null> {
+		return this.findOne(
+			{
+				tmid,
+			},
+			{
+				sort: {
+					ls: 1,
```

Comment:
A sort without an index is always a bad idea ;)

Reply:
Index added. I had one index there already, but I guess it's best to keep this as a separate one, right?